File: Programing C++/lab1/py/func.py
import sys
def inputText():
    print("Enter text. To finish press ctrl+z")
    text = sys.stdin.read()
    text[:len(text)-1]
    return text
def writeTextFile(text, textFileName):
    textFile = open(textFileName, 'wt')
    textFile.write(text)
    textFile.close()
def readTextFile(textFileName):
    textFile = open(textFileName, "rt")
    text = textFile.read()
    textFile.close()
    text = text[:len(text)-1]
    return text
def changeText(text):
    changedText = ''
    textLines = text.split('\n')
    for textLine in textLines:
        for i in range(len(textLine)):
            if textLine[i] == '1':
                changedText += '0'
            elif textLine[i] == '0':
                changedText += '1'
            else:
                changedText += textLine[i]
        changedText += '\n'
    return changedText
def outputTextFile(textFileName):
    textFile = open(textFileName, 'rt')
    text = textFile.read()
    textFile.close()
    print(text)

File: Programing C++/lab1/cpp/func.cpp
#include "func.h"
vector<string> inputText() 
{
	cout << "Enter text. To finish press ctrl+q" << endl;
	vector <string> text;
	string textLine;
	int code = 17;
	while (int(textLine[0]) != code) 
	{
		getline(cin, textLine);
		text.push_back(textLine);
	}
	text.pop_back();
	return text;
}
void writeTextFile(vector<string> text, string textFileName)
{
	ofstream textFile(textFileName);
	for (int i = 0; i < text.size(); i++)
	{
		textFile << text[i] << endl;
	}
	textFile.close();
}
vector<string> readTextFile(string textFileName) 
{
	vector<string> text;
	string textLine;
	ifstream textFile(textFileName);
	while (getline(textFile, textLine))
	{
		text.push_back(textLine);
	}
	textFile.close();
	return text;
}
vector<string> changeText(vector<string> text)
{
	vector<string> changedText;
	string changedTextLine = "";
	for (int i = 0; i < text.size(); i++)
	{
		for (int j = 0; j < text[i].length(); j++)
		{
			if (text[i][j] == '1')
			{
				changedTextLine += "0";
			}
			else if (text[i][j] == '0')
			{
				changedTextLine += "1";
			}
			else
			{
				changedTextLine += text[i][j];
			}
		}
		changedText.push_back(changedTextLine);
		changedTextLine = "";
	}
	return changedText;
}
void outputTextFile(string textFileName) 
{
	ifstream textFile(textFileName);
	string textLine;
	while (getline(textFile, textLine))
	{
		cout << textLine << endl;
	}
	textFile.close();
}